### PR TITLE
kill all tokens in the parser for good

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -489,7 +472,7 @@ def _parse_select(self):
             this = self.expression(
                 exp.Select,
                 hint=self._parse_hint(),
-                distinct=bool(self._match(TokenType.DISTINCT)),
+                distinct=self._match(TokenType.DISTINCT),
```
Comment (izeigerman) [inline sqlglot/parser.py:475]: Why isn't `bool` required here?

File: sqlglot/parser.py [hunk 2]
```diff
@@ -668,18 +655,14 @@ def _parse_order(self):
...
-        )
+        self._match(TokenType.ASC)
+        return self.expression(exp.Ordered, this=this, desc=self._match(TokenType.DESC))
 
     def _parse_limit(self):
         if not self._match(TokenType.LIMIT):
             return None
 
-        limit_number = self._match(TokenType.NUMBER)
+        limit_number = self._match(TokenType.NUMBER) and self._prev
```
Comment (izeigerman) [inline sqlglot/parser.py:665]: Given that we previously matched the `LIMIT` token how can `self._prev` be None in this case?

Comment (izeigerman) [reply sqlglot/parser.py:665]: Never mind. I confused it with `_curr`.

File: sqlglot/parser.py [hunk 3]
```diff
@@ -903,28 +888,52 @@ def _parse_column(self):
...
             return self._parse_case()
-        if not self._match(TokenType.L_PAREN):
-            return this
 
-        if this.token_type == TokenType.CAST:
+        if self._match(TokenType.CAST):
+            self._match(TokenType.L_PAREN)
             this = self._parse_cast()
-        elif this.token_type == TokenType.COUNT:
+            self._match(TokenType.R_PAREN)
```
Comment (izeigerman) [inline sqlglot/parser.py:898]: Here and in branches below: should we actually check whether the match was successful?

Comment (izeigerman) [reply sqlglot/parser.py:898]: Wouldn't the lack of right parenthesis be considered a syntax error?

File: sqlglot/parser.py [hunk 4]
```diff
@@ -327,10 +314,10 @@ def _parse_exists(self, not_=False):
         )
 
     def _parse_create(self):
-        temporary = bool(self._match(TokenType.TEMPORARY))
-        replace = bool(self._match(TokenType.OR) and self._match(TokenType.REPLACE))
+        temporary = self._match(TokenType.TEMPORARY)
+        replace = self._match(TokenType.OR) and self._match(TokenType.REPLACE)
```
Comment (izeigerman) [inline sqlglot/parser.py:318]: Side question: do we support `IF NOT EXISTS`?

Comment (tobymao) [reply sqlglot/parser.py:300]: yes, look at identity.sql line 232

**Pull request merged**